lib/src/main/scala/spinal/lib/Stream.scala

```diff
@@ -866,10 +866,23 @@ object StreamMux {
     c.io.output
   }
 
-  def apply[T <: Data](select: Stream[UInt], inputs: Vec[Stream[T]]): Stream[T] = {
+  /** syncSel joins the selection stream with the the output stream.
+    * Making sure the selection stream is fully synchronized with the selected stream
+    */
+  def syncSel[T <: Data](select: Stream[UInt], inputs: Seq[Stream[T]]): Stream[T] = {
```

Comment:
I mean this function name and the regSel also.

Reply:
Now I am confused. Those are the names proposed by @Dolu1990 in the issue (#1307)
Can you propose alternative names, which make it more clear?

Reply:
@g0t00 Yes right.
But now i'm seeing it, i'm thinking maybe the best would be to have it named : 
joinSel

So, we would have joinSel and regSel

I think the issue was that syncSel was too abstract, sync can mean many things.
But joinSel is very explicit :)
Sorry for the change.

Reply:
Seems good to me actually prefer that.